### DISCUSSION TITLE
Smart Stacks

### DISCUSSION THREAD

**Discussion by bkonia** [stacks-task-manager/stacks2#34] [Ideas & Feature requests]
It would be great if Stacks supported “Smart Stacks”, which would allow for creating a dynamic stack that would populate with tasks based on a query. For example, you could create an “Overdue” stack containing all tasks that are past their due date. The existing automation rules could be repurposed to be applied dynamically

**Comment by synapse** [comment]
Hey @bkonia the list view has several grouping including overdue tasks. 
Having a column based on a query would not make much sense since you might see (more or less) the same tasks in different columns. Another problem with this is that users might intend to move/drag tasks across columns (including from and to these `smart columns`) and the app would not know how to deal with this kind of interaction.

**Comment by bkonia** [comment]
> you might see (more or less) the same tasks in different columns.

Absolutely, and that would be a feature, not a bug. Many high-end project and task management systems allow for tasks to appear in multiple locations throughout the hierarchy and/or multiple columns in a Kanban view. This is far more powerful and flexible than the old fashioned paradigm of one location per task. We're not using paper planners anymore, so it's easy to create aliases that provide different views of the same data.

> users might intend to move/drag tasks across columns (including from and to these smart columns) and the app would not know how to deal with this kind of interaction.

Sure it would, because you'd program it to understand that tasks cannot be dragged to or from Smart Stacks. When attempting to drag a task out of a smart stack, the task would not be selectable. When attempting to drag to a smart stack, the cursor would change to a circle with a slash, indicating that the smart stack is not a valid drop point.
